lua/rustaceanvim/lsp/init.lua

```diff
@@ -285,10 +255,14 @@ end
 
 ---Stop the LSP client.
 ---@param bufnr? number The buffer number, defaults to the current buffer
+---@param exclude_rustc_target? string Cargo target triple (e.g., 'x86_64-unknown-linux-gnu') to filter rust-analyzer clients
 ---@return vim.lsp.Client[] clients A list of clients that will be stopped
-M.stop = function(bufnr)
+M.stop = function(bufnr, exclude_rustc_target)
   bufnr = bufnr or vim.api.nvim_get_current_buf()
-  local clients = rust_analyzer.get_active_rustaceanvim_clients(bufnr)
+  local clients = rust_analyzer.get_active_rustaceanvim_clients(
+    bufnr,
+    { exclude_rustc_target = exclude_rustc_target or DEFAULT_RUSTC_TARGET }
```

Comment:
issue: The `stop` function shouldn't set a rustc target by default, otherwise it will always try to filter, even when we don't need it to. 

```suggestion
    { exclude_rustc_target = exclude_rustc_target }
```

Reply:
yeah make sense. thanks